src/Product/list.ts

```diff
@@ -2,32 +2,22 @@ import { HttpModule } from '@whppt/api-express';
 import { Product } from './Models/Product';
 
 const list: HttpModule<{ domainId: string; limit: string; currentPage: string; search: string }, { products: Product[]; total: number }> = {
-  authorise() {
-    return Promise.resolve();
-  },
-  exec({ $mongo: { $dbPub } }, { domainId, limit, currentPage, search }) {
+  exec({ $database }, { domainId, limit, currentPage, search }) {
     let query = {
       domainId,
     } as any;
-
     if (search) {
       query = {
         $and: [{ domainId }, { $or: [{ name: { $regex: search, $options: 'i' } }, { productCode: { $regex: search, $options: 'i' } }] }],
```

Comment:
This might get slow with a lot of products.

What is the difference between list and filter. Could we include the search in the filter optionally?